src/Plotting/StrengthEnvelope.jl

```diff
@@ -0,0 +1,319 @@
+        iter += 1
+
+        # update densities
+        args  = (T=T, P=P)
+        compute_density!(ρ, MatParam, Phases, args)
+
+        # update pressure
+        for i = 2 : nz
+            P[i] = P[i-1] + ρ[i-1] * g * dz
+        end
+
+        # compute new norm
+        n    = (sum(P.^2)).^0.5
+
+        # evaluate stopping criterium
+        stop = abs((n-nprev) / (n+nprev))
+    end
+
+    #println("Finding lithostatic pressure took $iter iterations.")
+    return P
+end
+
+function solveStress(MatParam, Phases, ε, P, T)
+    # solve for stress
+    nz = length(T)
+    τ  = zeros(Float64, nz)
+    for i = 1 : nz
+        Pres = P[i]
+        Temp = T[i]
+        args = (T=Temp, P=Pres)
+        Mat  = MatParam[Phases[i]]
+        τ[i] = compute_τII(Mat.CreepLaws[1], ε, args)
+
+        F    = compute_yieldfunction(Mat.Plasticity[1], P=Pres, τII=τ[i])
+        if F > 0
+            ϕ = Mat.Plasticity[1].ϕ.val
+            c = Mat.Plasticity[1].C.val
+            τ[i] = Pres * sind(ϕ) + cosd(ϕ) * c
+        end
+    end
+
+    return τ
+end
+
+function redimensionalize(x, nz, param_dim::Unitful.FreeUnits, CharDim)
+    x_val = zeros(Float64, nz)
+    x_dim = dimensionalize(x, param_dim, CharDim)
+    for i = 1 : nz
+        x_val[i] = x_dim[i].val
+    end
+    return x_val
+end
+
+"""
+    StrengthEnvelopeSliders(MatParam, Thickness, TempType=LinearTemp()::AbstractThermalStructure)
+
+Creates a GUI that plots a 1D strength envelope. In the GUI, temperature profile and strain rate can be adjusted. The Drucker-Prager plasticity uses lithostatic pressure.
+
+Parameters:
+- MatParam:  a tuple of materials (including the following properties: Phase, Density, CreepLaws, Plasticity)
+- Thickness: a vector listing the thicknesses of the respective layers (should carry units)
+- TempType:  the type of temperature profile (LinearTemp=default, HalfspaceCoolingTemp, ConstantTemp)
+
+# Example:
+```julia-repl
+julia> using GLMakie
+julia> MatParam = (SetMaterialParams(Name="UC", Phase=1, Density=ConstantDensity(ρ=2700kg/m^3), CreepLaws = SetDislocationCreep("Wet Quartzite | Ueda et al. (2008)"), Plasticity = DruckerPrager(ϕ=30.0, C=10MPa)),
+                   SetMaterialParams(Name="MC", Phase=2, Density=Density=ConstantDensity(ρ=2900kg/m^3), CreepLaws = SetDislocationCreep("Plagioclase An75 | Ji and Zhao (1993)"), Plasticity = DruckerPrager(ϕ=20.0, C=10MPa)),
+                   SetMaterialParams(Name="LC", Phase=3, Density=PT_Density(ρ0=2900kg/m^3, α=3e-5/K, β=1e-10/Pa), CreepLaws = SetDislocationCreep("Maryland strong diabase | Mackwell et al. (1998)"), Plasticity = DruckerPrager(ϕ=30.0, C=10MPa)));
+julia> Thickness = [15,10,15]*km;
+
+julia> StrengthEnvelope(MatParam, Thickness, LinearTemp())
+```
+"""
+function StrengthEnvelope(MatParam::NTuple{N, AbstractMaterialParamsStruct}, Thickness::Vector{U}, TempType::AbstractThermalStructure=LinearTemp(), mode::String="normal") where {N, U}
+
+    # hardcoded input
+    nz        = 101
```

Comment:
can you make this a changeable part of the input function?
Do that with:
```Julia
function StrengthEnvelope(MatParam::NTuple{N, AbstractMaterialParamsStruct}, Thickness::Vector{U}, TempType::AbstractThermalStructure=LinearTemp(), mode::String="normal"; nz=101)
```
Default will be 101 in that case, but you have the possibility to increase it

Reply:
> > this is nice very nice. Yet, it would be fantastic to not just restrict this to linear temperature structures, but instead also allow half-space cooling thermal structures (with given age; perhaps as slider?) Could that be added?
> 
> That is already part of it

Sorry, I had missed that.